roles/common/tasks/npm_install.yml

```diff
@@ -1,13 +1,49 @@
 ---
 # Node.js and npm installation tasks
 
-- name: Install Node.js and npm via system package manager
-  ansible.builtin.package:
-    name:
-      - nodejs
-      - npm
-    state: present
-  become: true
+- name: "Check if Node.js is already installed (minimum version: {{ common_nodejs_min_version }})"
+  ansible.builtin.shell: |
+    set -o pipefail
+    if command -v node >/dev/null 2>&1; then
+      node_version=$(node --version | cut -d'v' -f2 | cut -d'.' -f1)
+      if [ "$node_version" -ge {{ common_nodejs_min_version }} ]; then
+        echo "sufficient"
+      else
+        echo "insufficient"
+      fi
+    else
+      echo "not_installed"
+    fi
+  register: common_node_check
+  changed_when: false
+
+- name: "Install Node.js from NodeSource (version: {{ common_nodejs_min_version }})"
+  when: common_node_check.stdout != "sufficient"
+  block:
+    - name: Download NodeSource setup script
+      ansible.builtin.get_url:
+        url: https://rpm.nodesource.com/setup_{{ common_nodejs_min_version }}.x
+        dest: /tmp/nodesource_setup.sh
+        mode: "0755"
+      become: true
+
+    - name: Run NodeSource setup script
+      ansible.builtin.command:
+        cmd: bash /tmp/nodesource_setup.sh
+      become: true
+      changed_when: true
+
+    - name: "Install Node.js package (version: {{ common_nodejs_min_version }})"
```

Comment:
If another npm MCP server has already been installed and a lower version of Node is already installed, does `ansible.builtin.package` override the existing version and install the required newer one from NodeSource?